cassandra/src/main/java/com/yahoo/ycsb/db/CassandraCQLClient.java

```diff
@@ -425,30 +526,34 @@ public Status insert(String table, String key,
       HashMap<String, ByteIterator> values) {
 
     try {
-      Insert insertStmt = QueryBuilder.insertInto(table);
+      final int numValues = values.size();
+      int i = 0;
+      Object[] vals = new Object[numValues + 1];
 
       // Add key
-      insertStmt.value(YCSB_KEY, key);
+      vals[i++] = key;
 
-      // Add fields
-      for (Map.Entry<String, ByteIterator> entry : values.entrySet()) {
-        Object value;
-        ByteIterator byteIterator = entry.getValue();
-        value = byteIterator.toString();
-
-        insertStmt.value(entry.getKey(), value);
+      // Add fieldList
+      for (; i < fieldList.size(); i++) {
+        String field = fieldList.get(i);
+        if (values.containsKey(field)) {
+          vals[i] = values.get(field).toString();
+        }
       }
 
-      insertStmt.setConsistencyLevel(writeConsistencyLevel);
+      // If there is one value, this is actually an update
+      BoundStatement bs = (numValues == 1 ?
+               updateStatements.get(values.keySet().iterator().next()) :
```

Comment:
Hmm. Good catch. Not sure why I didn't I didn't see that in IntelliJ. 

I feel like I did have it at one point before I attempted to re-order the "sections" of statement initializations. 
https://github.com/jbellis/YCSB/blob/217020bd07e556ba2870e2487435d6297e18fb3c/cassandra/src/main/java/com/yahoo/ycsb/db/CassandraCQLClient.java#L174-L185